src/ansys/fluent/core/session.py

```diff
@@ -343,8 +345,49 @@ def download(self, file_name: str, local_file_name: Optional[str] = None):
         FileNotFoundError
             If the remote file does not exist.
         """
-        if self.file_service:
-            if self.file_service.file_exist(file_name):
-                self.file_service.download_file(file_name, local_file_name)
+        self._upload_download_helper(
+            is_upload=False, file_name=file_name, local_file_name=local_file_name
+        )
+
+    def _pypim_upload(self, file_name: str, on_uploaded: Optional[Any] = None):
+        """Uploads a file if not available on the server.
+
+        Parameters
+        ----------
+        file_name : str
+            File name
+        on_uploaded: Callable[str]
+            Read a file.
+        Raises
+        ------
+        FileNotFoundError
+            If a file does not exist.
+        """
+        if pypim.is_configured():
+            if os.path.isfile(file_name):
+                if not self.file_service.file_exist(os.path.basename(file_name)):
+                    self.upload(file_name)
+            elif self.file_service.file_exist(os.path.basename(file_name)):
+                pass
+            else:
+                raise FileNotFoundError(f"{file_name} does not exist.")
```

Comment:
rather than 
```
if ...:
  ...
elif ...:
  pass
else:
  ...
```
prefer:
```
if ...:
  ...
elif not ...:
  ...
```

Reply:
It's not clear to me about `elif not ...`;

Reply:
We don't have `elif not` scenario in above code block.

Reply:
I see it. Its equivalent.

Reply:
Current code implemented according to following discussion.

![image](https://github.com/ansys/pyfluent/assets/106588300/3d99be07-6b6c-4634-b011-b354f78504f7)

Reply:
We can do following change. 

```
if self.file_service.file_exist(os.path.basename(file_name)):
	pass
elif not self.file_service.file_exist(os.path.basename(file_name)):
	try:
		if os.path.isfile(file_name):
			self.upload(file_name)
	except FileNotFoundError as e:
		print(e.message)
else:
	raise FileNotFoundError(f"{file_name} does not exist.")
```
 OR
 
```
 if self.file_service.file_exist(os.path.basename(file_name)):
	pass
elif not self.file_service.file_exist(os.path.basename(file_name)):
	if os.path.isfile(file_name):
		self.upload(file_name)
	else:
		raise FileNotFoundError(f"{file_name} does not exist.")
else:
	raise FileNotFoundError(f"{file_name} does not exist.")
```

We need to raise an error or exception 2 times if we follow your suggestion and I think existing code handles this already.

@seanpearsonuk Could you please explain how your suggestion improves current code if it is equivalent, if I missing any point in existing code ?